drivers/watchdog/wdt_rpi_pico.c

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright (c) 2022, Jamie McCrae
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT raspberrypi_pico_watchdog
+
+#include <hardware/watchdog.h>
+#include <hardware/structs/psm.h>
+#include <drivers/watchdog.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(wdt_rpi_pico, CONFIG_WDT_LOG_LEVEL);
+
+/* Maximum watchdog time is halved due to errata RP2040-E1 */
+#define RPI_2040_MAX_WDT_TIME (0xffffff / 2)
+#define RPI_2020_WDT_TIME_MULTIPLCATION_FACTOR 2
+
+/* Watchdog requires a 1MHz clock source, divided from the crystal oscillator */
+#define RPI_2040_XTAL_FREQ_WDT_TICK_DIVISOR 1000000
```

Comment:
So, this is the target frequency, right? Not the divisor. This is used to calculate the divider for the particular clock speed.

Reply:
It is both, the target frequency is 1MHz but the tick count is in cycles, so at a clock speed of 12MHz, this needs to be divided by 1M to get 12, hence the name, as the tick count is fed from the 1MHz timer source.